src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1232,461 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
+        {
+            await Task.Run(async () =>
+            {
+                var result = new ResultStatus();
+                try
+                {
+
+                    string deleteNotebookRowQuery =
+                    @"
+                    DELETE FROM notebooks.nb_template
+                    WHERE 
+                    job_id = @jobId;
+                    DELETE FROM notebooks.nb_materialized
+                    WHERE
+                    job_id = @jobId;
+                    IF NOT EXISTS (SELECT * FROM notebooks.nb_template)
+                    BEGIN
+                        DROP TABLE notebooks.nb_template;
+                        DROP TABLE notebooks.nb_materialized;
+                        DROP SCHEMA notebooks;
+                    END
+                    ";
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(parameters.OwnerUri, out connInfo);
+                    List<SqlParameter> deleteNotebookRowQueryParams = new List<SqlParameter>();
+                    deleteNotebookRowQueryParams.Add(new SqlParameter("jobId", parameters.Notebook.JobId));
+                    ExecuteQuery(connInfo, deleteNotebookRowQuery, deleteNotebookRowQueryParams, parameters.Notebook.TargetDatabase);
+                    await ConfigureAgentJob(
+                        parameters.OwnerUri,
+                        parameters.Notebook.Name,
+                        parameters.Notebook,
+                        ConfigAction.Drop,
+                        ManagementUtils.asRunType(parameters.TaskExecutionMode));
+
+                    result.Success = true;
+                }
+                catch (Exception e)
+                {
+                    result.Success = false;
+                    result.ErrorMessage = e.ToString();
+                }
+                await requestContext.SendResult(result);
+            });
+        }
+
+        internal async Task HandleUpdateAgentNotebookRequest(UpdateAgentNotebookParams parameters, RequestContext<UpdateAgentNotebookResult> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                var result = new UpdateAgentNotebookResult();
+                try
+                {
+
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                                                parameters.OwnerUri,
+                                                out connInfo);
+                    string targetDatabase = parameters.Notebook.TargetDatabase;
+                    string jobId = parameters.Notebook.JobId;
+                    if (parameters.TemplateFilePath != null)
+                    {
+
+                        string templateFilePath = parameters.TemplateFilePath;
+                        string templateFileContents = File.ReadAllText(templateFilePath);
+                        templateFileContents = templateFileContents.Replace("'", "''");
```

Comment:
I don't think you need to escape the `'` characters if you are using SqlParameters.